core/src/main/scala/com/criteo/cuttle/Authentication.scala

```diff
@@ -0,0 +1,89 @@
+package com.criteo.cuttle
+
+import java.util.Base64
+
+import lol.http._
+
+import scala.util.Try
+
+/**
+  * Trait describing authentication methods
+  */
+trait Authenticator {
+  /**
+    *
+    * @param wrappedAuthedService
+    * @return
+    */
+  def apply(wrappedAuthedService: (User) => Service): Service = (request: Request) => {
+    authenticate(request)
+      .fold(user => wrappedAuthedService(user)(request), identity)
+  }
+
+  /**
+    * Performs authentication on request.
+    * @param r the request to be authenticated
+    * @return either an authenticated user or a response.
+    */
+  def authenticate(r: Request): Either[User, Response]
```

Comment:
Just a matter of style: we usually use Right for the successful value and in this case it is most likely the User. So I would inverse both types.